src/segments/argocd_test.go

```diff
@@ -0,0 +1,233 @@
+package segments
+
+import (
+	"fmt"
+	"github.com/jandedobbeleer/oh-my-posh/src/mock"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/stretchr/testify/assert"
+	mock2 "github.com/stretchr/testify/mock"
+	"path"
+	"testing"
+)
+
+func TestArgocdGetConfigFromOpts(t *testing.T) {
+	configFile := "/Users/posh/.configFile/argocd/config"
+	cases := []struct {
+		Case          string
+		Opts          string
+		Expected      string
+		ExpectedError string
+	}{
+		{Case: "invalid flag in opts", Opts: "--invalid", ExpectedError: argocdInvalidFlag},
+		{Case: "another invalid flag in opts", Opts: "-invalid", ExpectedError: argocdInvalidFlag},
+		{Case: "no config in opts", Opts: "--grpc-web", Expected: ""},
+		{Case: "config in opts", Opts: fmt.Sprintf("--grpc-web --config %s --plaintext", configFile), Expected: configFile},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("Getenv", argocdOptsEnv).Return(tc.Opts)
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		if len(tc.ExpectedError) > 0 {
+			_, err := argocd.getConfigFromOpts()
+			assert.EqualError(t, err, tc.ExpectedError, tc.Case)
+		} else {
+			config, err := argocd.getConfigFromOpts()
+			assert.NoErrorf(t, err, tc.Case)
+			assert.Equal(t, tc.Expected, config, tc.Case)
+		}
+	}
+}
+
+func TestArgocdGetConfigPath(t *testing.T) {
+	home := "/Users/posh"
+	configFile := path.Join(home, ".config", "argocd", "config")
+	cases := []struct {
+		Case          string
+		Opts          string
+		Expected      string
+		ExpectedError string
+	}{
+		{Case: "without opts", Expected: configFile},
+		{Case: "with opts", Opts: "--config /etc/argocd/config", Expected: "/etc/argocd/config"},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.MockedEnvironment)
+		env.On("Home").Return(home)
+		env.On("Getenv", argocdOptsEnv).Return(tc.Opts)
+
+		argocd := &Argocd{
+			env:   env,
+			props: properties.Map{},
+		}
+		assert.Equal(t, tc.Expected, argocd.getConfigPath())
+	}
+}
+
+func TestArgocdParseConfig(t *testing.T) {
+	configFile := "../test/argocd.yml"
```

Comment:
this doesn't exist, right?

Reply:
correct. it's just some kind of placeholder here. because we may need to parse different configs in this UT.

Reply:
updated with the same config file like other UT.